src/promptflow/promptflow/_cli/_pf/_flow.py

```diff
@@ -423,33 +437,74 @@ def test_flow(args):
 
 
 def serve_flow(args):
+    from promptflow._sdk._load_functions import load_flow
+
     logger.info("Start serve model: %s", args.source)
     # Set environment variable for local test
     source = Path(args.source)
-    os.environ["PROMPTFLOW_PROJECT_PATH"] = source.absolute().as_posix()
-    from promptflow._sdk._serving.app import create_app
-
-    static_folder = args.static_folder
-    if static_folder:
-        static_folder = Path(static_folder).absolute().as_posix()
     logger.info(
         "Start promptflow server with port %s",
         args.port,
     )
-    config = list_of_dict_to_dict(args.config)
-    # Change working directory to model dir
-    print(f"Change working directory to model dir {source}")
-    os.chdir(source)
-    app = create_app(
-        static_folder=static_folder,
-        environment_variables=list_of_dict_to_dict(args.environment_variables),
-        config=config,
-    )
-    target = f"http://{args.host}:{args.port}"
-    print(f"Opening browser {target}...")
-    webbrowser.open(target)
-    # Debug is not supported for now as debug will rerun command, and we changed working directory.
-    app.run(port=args.port, host=args.host)
+    os.environ["PROMPTFLOW_PROJECT_PATH"] = source.absolute().as_posix()
+    flow = load_flow(args.source)
+    if "language" in flow.dag and flow.dag["language"] == FlowLanguage.CSharp:
+        from promptflow.batch._csharp_executor_proxy import EXECUTOR_SERVICE_DLL
```

Comment:
may create two functions for these two languages

Reply:
updated